packages/codegen-ui-react/lib/react-table-renderer.ts

```diff
@@ -0,0 +1,274 @@
+  createTableHeadElementBlock(): JsxExpression {
+    return factory.createJsxExpression(
+      undefined,
+      factory.createBinaryExpression(
+        factory.createPrefixUnaryExpression(
+          SyntaxKind.ExclamationToken,
+          factory.createJsxExpression(undefined, factory.createIdentifier('disableHeaders')),
+        ),
+        SyntaxKind.AmpersandAmpersandToken,
+        this.createTableHeadElement(),
+      ),
+    );
+  }
+
+  createRowOnClickCBAttr(): JsxAttributes {
+    return factory.createJsxAttributes([
+      factory.createJsxAttribute(
+        factory.createIdentifier('onClick'),
+        factory.createJsxExpression(
+          undefined,
+          factory.createConditionalExpression(
+            factory.createBinaryExpression(
+              factory.createIdentifier('highlightOnHover'),
+              factory.createToken(SyntaxKind.AmpersandAmpersandToken),
+              factory.createIdentifier('onRowClick'),
+            ),
+            factory.createToken(SyntaxKind.QuestionToken),
+            factory.createArrowFunction(
+              undefined,
+              undefined,
+              [],
+              undefined,
+              factory.createToken(SyntaxKind.EqualsGreaterThanToken),
+              factory.createCallExpression(factory.createIdentifier('onRowClick'), undefined, [
+                factory.createIdentifier('item'),
+                factory.createIdentifier('index'),
+              ]),
+            ),
+            factory.createToken(SyntaxKind.ColonToken),
+            factory.createNull(),
+          ),
+        ),
+      ),
+    ]);
+  }
+
+  createTableBodyCellFromColumn(column: ColumnInfo): JsxElement {
+    const columnId = column.header;
+
+    return factory.createJsxElement(
+      factory.createJsxOpeningElement(
+        factory.createIdentifier(Primitive.TableCell),
+        undefined,
+        factory.createJsxAttributes([]),
+      ),
+      [
+        factory.createJsxExpression(
+          undefined,
+          factory.createConditionalExpression(
+            factory.createElementAccessExpression(
+              factory.createIdentifier('format'),
+              factory.createStringLiteral(columnId),
+            ),
+            factory.createToken(SyntaxKind.QuestionToken),
+            factory.createCallExpression(
+              factory.createElementAccessExpression(
+                factory.createIdentifier('format'),
+                factory.createStringLiteral(columnId),
+              ),
+              undefined,
+              [
+                factory.createPropertyAccessExpression(
+                  factory.createIdentifier('item'),
+                  factory.createIdentifier(columnId),
+                ),
+              ],
+            ),
+            factory.createToken(SyntaxKind.ColonToken),
+            factory.createPropertyAccessExpression(
+              factory.createIdentifier('item'),
+              factory.createIdentifier(columnId),
+            ),
```

Comment:
Don't we want to pass the item, the format type, and the formatting spec?

Reply:
Need to rethink this one since customers can provide their own formatting by passing override, we also have configured formatting from CP, and no formatting (literal value).

Reply:
Ah okay let's add changes as a follow up